BibBuddy/app/src/main/java/de/bibbuddy/NotesFragment.java

```diff
@@ -25,103 +24,96 @@
   private SortCriteria sortCriteria;
+  private TextView emptyListView;
 
   @Nullable
   @Override
   public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                            @Nullable Bundle savedInstanceState) {
-    requireActivity().getOnBackPressedDispatcher().addCallback(new OnBackPressedCallback(true) {
-      @Override
-      public void handleOnBackPressed() {
-
-        FragmentManager fm = getParentFragmentManager();
-        if (fm.getBackStackEntryCount() > 0) {
-          fm.popBackStack();
-        } else {
-          requireActivity().onBackPressed();
-        }
-      }
-    });
-
     View view = inflater.inflate(R.layout.fragment_notes, container, false);
-    recyclerView = view.findViewById(R.id.recyclerView);
-
-    sortCriteria = ((MainActivity) getActivity()).getSortCriteria();
 
     noteModel = new NoteModel(getContext());
-    notes = noteModel.getCompleteNoteList();
-    setHasOptionsMenu(true);
-    setupRecyclerView();
-    enableSwipeToDelete();
+    notesRecyclerView = view.findViewById(R.id.note_list_recycler_view);
+    noteList = noteModel.getNoteList();
+    sortCriteria = ((MainActivity) requireActivity()).getSortCriteria();
+    emptyListView = view.findViewById(R.id.empty_notes_list_view);
 
-    ((MainActivity) getActivity()).setVisibilityImportShareButton(View.INVISIBLE, View.INVISIBLE);
+    ((MainActivity) requireActivity())
+        .setVisibilityImportShareButton(View.INVISIBLE, View.INVISIBLE);
     setupSortBtn();
+    setHasOptionsMenu(true);
+    setupRecyclerView();
 
     return view;
   }
 
-  private void setupSortBtn() {
-    ImageButton sortBtn = getActivity().findViewById(R.id.sort_btn);
-    ((MainActivity) getActivity()).setVisibilitySortButton(true);
-
-    sortBtn.setOnClickListener(new View.OnClickListener() {
-      @Override
-      public void onClick(View v) {
-        handleSortNote();
-      }
-    });
-  }
-
   @Override
   public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
     inflater.inflate(R.menu.fragment_note_list_menu, menu);
     super.onCreateOptionsMenu(menu, inflater);
   }
 
+  @Override
+  public void onPrepareOptionsMenu(Menu menu) {
+    MenuItem deleteNote = menu.findItem(R.id.menu_note_list_delete);
+    deleteNote.setVisible(adapter.getSelectedNoteItems().size() > 0);
+  }
 
   @Override
   public boolean onOptionsItemSelected(MenuItem item) {
-
-    switch (item.getItemId()) {
-
-      case R.id.menu_help_note_list:
-        handleManualNotesList();
-        break;
-
-      default:
-        Toast.makeText(getContext(), "Fehler", Toast.LENGTH_SHORT).show();
+    long id = item.getItemId();
+    if (id == R.id.menu_note_list_delete) {
+      handleDeleteNote();
+    } else if (item.getItemId() == R.id.menu_note_list_help) {
+      handleHelpNotesFragment();
+    } else {
+      Toast.makeText(getContext(), R.string.error, Toast.LENGTH_SHORT).show();
```

Comment:
I think we should provide a more meaningful error here - this will probably only create confusion.

Reply:
This error texts as default else-case for options-menu are legacy and removed from BookNotesView and NotesFragment at least.